aQute.libg/src/aQute/lib/concurrent/serial/SerialExecutor.java

```diff
@@ -0,0 +1,90 @@
+package aQute.lib.concurrent.serial;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.Executor;
+
+import org.osgi.util.promise.Deferred;
+import org.osgi.util.promise.Promise;
+
+/**
+ * Will execute a set of tasks in order of submit.
+ */
+public class SerialExecutor implements Closeable {
+	final Executor			executor;
+	final List<Runnable>	tasks	= new ArrayList<>();
```

Comment:
I assume you really want to use  a Deque here rather than list.

Reply:
No, why? I need so synchronize so there is not a lot of difference? Lists are also quite small.

Reply:
ok